app/assets/javascripts/admin/order_cycles/controllers/edit.js.coffee

```diff
@@ -18,5 +19,14 @@ angular.module('admin.orderCycles')
 
     $scope.submit = ($event, destination) ->
       $event.preventDefault()
+      $scope.navClear()
```

Comment:
Did you try just calling `NavigationCheck.clear()` here? I'm not sure if you need to expose that function to the scope.